pkg/cmd/pr/create/create.go

```diff
@@ -134,31 +141,32 @@ func NewCmdCreate(f *cmdutil.Factory, runF func(*CreateOptions) error) *cobra.Co
 	fl.StringSliceVarP(&opts.Labels, "label", "l", nil, "Add labels by `name`")
 	fl.StringSliceVarP(&opts.Projects, "project", "p", nil, "Add the pull request to projects by `name`")
 	fl.StringVarP(&opts.Milestone, "milestone", "m", "", "Add the pull request to a milestone by `name`")
+	fl.StringVarP(&opts.JSONInput, "json", "j", "", "Use JSON to populate and submit PR")
```

Comment:
Would we consider hiding this flag from docs to avoid signaling to users that this flag and the related JSON format aren't part of the public API for gh? This would be to allow changing parts of this mechanism without fearing breaking people's scripts.

Reply:
I was thinking this might be of use _as_ part of the public API for use in scripting. When you  suggest hiding it, do you mean in perpetuity or until we've discussed it more?

Reply:
The way I see it, our public API to use for scripting is the flags themselves. This adds another one, essentially doubling the API surface for `issue/pr create` commands, since now every piece of information can be either passed through a CLI flag or through a JSON parameter.

When I suggested hiding it, I meant at least temporarily until the JSON API solidifies and we saw more use-cases for it other than the restore functionality.

Reply:
How would you feel about renaming it to `--recover` and having it only take a filepath argument without advertising it as supporting json?

Reply:
That sounds great!!